20_Simulation_Principles/50_Multi-Model_Representation/12_ForceField.md

```diff
@@ -19,69 +19,64 @@ ForceField API
+Using an [explicit scheme](../../components/odesolver/forward/eulerexplicitsolver/) means that forces $$f(x)$$ are computed using the degrees of freedom of the current time step $$t$$ (which are known): $$f(x)=f(x(t))$$. Regardless the form of the function $$f$$, the value of $$f(x(t))$$ can directly be obtained and set in the right hand side vector $$b$$ of our linear system $$\mathbf{A}x=b$$.
 
-The computation of the term <img class="latex" src="https://latex.codecogs.com/png.latex?f" title="Forces" />, the value of <img class="latex" src="https://latex.codecogs.com/png.latex?dt\cdot%20f(x(t))" title="Explicit forces" /> is done through the function `addForce()` of the ForceField class, called by the integration scheme (ODESolver).
+The computation of the term $$f$$, the value of $$dt\cdot%20f(x(t))$$ is done through the function `addForce()` of the ForceField class, called by the integration scheme (ODESolver).
 
 
 
 ### Implicit force
 
 
-Using an [implicit scheme](https://www.sofa-framework.org/community/doc/using-sofa/components/integrationscheme/eulerimplicitsolver/) means that forces <img class="latex" src="https://latex.codecogs.com/png.latex?f(x)" title="Forces" /> are computed using the degrees of freedom of the next time step <img class="latex" src="https://latex.codecogs.com/png.latex?t+dt" title="Current time" /> (unknown yet): <img class="latex" src="https://latex.codecogs.com/png.latex?f(x)=f(x(t+dt))" title="Explicit forces" />.
-The value of <img class="latex" src="https://latex.codecogs.com/png.latex?f(x(t))" title="Explicit forces" /> can not be directly be computed. By using a Taylor expansion, we get:
+Using an [implicit scheme](../../components/odesolver/backward/eulerimplicitsolver/) means that forces $$f(x)$$ are computed using the degrees of freedom of the next time step $$t+dt$$ (unknown yet): $$f(x)=f(x(t+dt))$$.
+The value of $$f(x(t))$$ can not be directly be computed. By using a Taylor expansion, we get:
 
-<img class="latex" src="https://latex.codecogs.com/png.latex?\mathbf{M}%20\Delta%20v=dt%20\cdot%20\left(%20f(x(t)+\cdot%20\frac{\partial%20f}{\partial%20x}%20\Delta%20x%20\right)" title="Implicit dynamic system" />
+$$\mathbf{M}%20\Delta%20v=dt%20\cdot%20\left(%20f(x(t)+\cdot%20\frac{\partial%20f}{\partial%20x}%20\Delta%20x%20\right)$$
+since we have: $$\Delta%20x=dt(v(t)+\Delta%20v)$$, then:
+$$\mathbf{M}\Delta%20v=dt\cdot%20\left(%20f(x(t)+dt\cdot%20\frac{\partial%20f}{\partial%20x}v(t)+dt\cdot%20\frac{\partial%20f}{\partial%20x}%20\Delta%20v%20\right)$$
 
-since we have: <img class="latex" src="https://latex.codecogs.com/png.latex?\Delta%20x=dt(v(t)+\Delta%20v)" title="Implicit scheme" />, then:
+Finally, gathering the unknown (depending on $$\Delta%20v$$) in the left hand side, we have:
```

Comment:
Shouldn't this '%20' be replaced by the real character ? It was introduced by the fact that it was in a url I guess. If so, there are multiple occurrence of this in the PR...

Reply:
No, the `%20` is a _ i guess. And you might want to modify it everywhere.

Reply:
Yes, now you might want to do it everywhere...